control/jet_vane_mapper.cc

```diff
@@ -0,0 +1,49 @@
+#include "control/jet_vane_mapper.hh"
+
+#include "third_party/experiments/numerics/group_diff.hh"
+
+namespace jet {
+namespace control {
+
+MappingResult JetVaneMapper::map_wrench(const Wrench& target_wrench, const JetStatus& current_jet_status) const {
+  using InformationMat = MatNd<QuadraframeStatus::DIM, QuadraframeStatus::DIM>;
+  using JacobianMat = MatNd<Wrench::DIM, QuadraframeStatus::DIM>;
+
+  // Wrench Error Jacobian
+  const auto held_function = [this, &current_jet_status](const QuadraframeStatus& qframe_status) -> Wrench {
+    return total_wrench_com_frame(current_jet_status, qframe_status, vane_cfg_, jet_cfg_, qframe_cfg_);
+  };
+
+  //
+  // Minimize
+  //
+
+  // Small levenberg damping to avoid numerical issues or unboundedness
+  constexpr double LEVENBERG_COEFF = 1e-3;
+
+  QuadraframeStatus cur_qframe_stat;
+  constexpr int MAX_ITERS = 5;
+  for (int k = 0; k < MAX_ITERS; ++k) {
```

Comment:
I have found that these sorts of optimizers are usually easier to interpret and implement inline; and writing a general one is usually more mess than its worth